src/pages/Dashboard/Profile.jsx

```diff
@@ -1,13 +1,294 @@
-import React from 'react';
+import React, { useCallback, useMemo, useReducer } from 'react';
+import styled from 'styled-components';
 import { AddressCard } from 'styled-icons/fa-regular/AddressCard';
 import { DashboardPage } from '../../components';
+import {
+    Button,
+    Card,
+    Checkbox,
+    Heading,
+    Input,
+    Select,
+} from '@cheapreats/react-ui';
+
+const testUser = {
+    status: 'PENDING',
+    email: 'jacky.chan@fake.utoronto.ca',
+    firstname: 'Jacky',
+    lastname: 'Chan',
+    gender: 'MALE',
+    school: 'UTSC',
+    bio: 'actor/fighter',
+    photo: 'https://i.ibb.co/HF6q3BF/doggo.gif',
+    links: [
+        {
+            name: 'github',
+            href: 'https://github.com/cheapreats/react-ui-library',
+        },
+        {
+            name: 'linkedin',
+            href: 'https://github.com/cheapreats/react-ui-library',
+        },
+        {
+            name: 'website',
+            href: 'https://github.com/cheapreats/react-ui-library',
+        },
+    ],
+};
+
+function reducer(state, action) {
+    switch (action.type) {
+        case 'update':
+            return { ...state, [action.name]: action.value };
+        case 'update_links':
+            const links = state.links.map(link =>
+                link.name === action.name
+                    ? { ...link, href: action.value }
+                    : link,
+            );
+            return { ...state, links };
+        default:
+            return state;
+    }
+}
 
 export const Profile = () => {
-    return <DashboardPage heading="Profile"></DashboardPage>;
+    const [user, dispatch] = useReducer(reducer, {
+        ...testUser,
+        currentPassword: '',
+        newPassword: '',
+        confirmPassword: '',
+    });
+    const onChange = useCallback(
+        e =>
+            dispatch({
+                type: 'update',
+                name: e.target.name,
+                value: e.target.value,
+            }),
+        [],
+    );
+    const onChangeLinks = useCallback(
```

Comment:
Similar to above, but add perhaps add an additional prop like link: [name of link]